### ISSUE TITLE
gl_FrontFace not working properly Adreno 530 on double sided materials (Samsung S7, OnePlus 3, Pixel, etc.)

### ISSUE THREAD

**Issue by bhouston** [mrdoob/three.js#10331]
##### Description of the problem 

So Qualcomm appears to have messed up gl_FrontFacing on a number of their GPUs when using double sided rendering.  It affects at least the Adreno 530 series of GPUs - which are used by most of the highest end phones around including Samsung Galaxy S7, OnePlus 3 (my current phone), and the Google Pixel.

I have a reproduction case in my latest PR: https://github.com/mrdoob/three.js/pull/10330

When using "front sided" on the EnvMap HDR example on Adren 530 you get this:

![image](https://cloud.githubusercontent.com/assets/588541/21077529/b4c140e4-bf1c-11e6-8705-066d6400e706.png)

When using "double sided" on the EnvMap HDR example on Adren 530 you get this (but it still looks like the above on desktop and on other GPUs) :

![image](https://cloud.githubusercontent.com/assets/588541/21077526/a27d9374-bf1c-11e6-842e-a0a377cff217.png)

After a ton of work I traced the issue to the flipNormal value here:

https://github.com/mrdoob/three.js/blob/dev/src/renderers/shaders/ShaderChunk/normal_flip.glsl#L2

Basically it seems that with Physical materials this gl_FrontFace is not working properly with double sided materials on Adreno GPUs.

##### Three.js version

- [x] Dev
- [x] r82

##### Browser

- [ ] All of them
- [x] Chrome
- [ ] Firefox
- [ ] Internet Explorer

##### OS

- [ ] All of them
- [ ] Windows
- [ ] Linux
- [x] Android
- [ ] IOS

##### Hardware Requirements (graphics card, VR Device, ...)

Adreno 5x0 and I believe Adreno 4x0 GPUs

**Comment by mrdoob** [comment]
In case anyone lands here, the Adreno bug is only when `gl_FrontFacing` is inside a function: #21205
